app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    stored_args_and_result = {}
+
+    def inner(*args) -> Any:
+
+        if args in stored_args_and_result:
+            print("Getting from cache")
+            return stored_args_and_result[args]
+
+        print("Calculating new result")
+        stored_args_and_result[args] = func(*args)
```

Comment:
No need to duplicate
```suggestion
        if args in stored_args_and_result:
            print("Getting from cache")
        else:
        print("Calculating new result")
        stored_args_and_result[args] = func(*args)
   return stored_args_and_result[args]
```